### ISSUE TITLE
About `dropout step` in the paper

### ISSUE THREAD

**Issue by luzai** [dstamoulis/single-path-nas#6]
Thank you very much for sharing the whole implementation! I am curious about `dropout step` in this figure, may I ask some questions? 
![image](https://user-images.githubusercontent.com/14788650/58747304-cfa4eb80-849b-11e9-983b-22cdcb8bab57.png)

- As shown in the code [here](https://github.com/dstamoulis/single-path-nas/blob/21c1f3e9c790e591749c0fb861bff3737b6d5fc7/nas-search/nas_utils.py#L56), I guess that dropout on indicator is applied during the warmup steps (6255 steps), the keep probability (rather than drop probability) is 0.6; after 6255 steps, the keep probability is 100, and the model will choose 3x3, 5x5, exp=3, exp=6 by learned threshold. May I ask whether my interpretation correct?
- May I ask why the runtime term is larger in the dropout steps? And why the runtime decrease rapidly and precisely to 79 ms? What whould the curve be like if hyperparameter lambda=0.02 is setted to other values?

**Comment by annamz6526-1** [comment]
Hi @luzai. I think your interpretation for the dropout is correct. But I couldn't figure out why they set the keep probability to 100, which means to scale the Indicator value 100 times smaller.

**Comment by luzai** [comment]
It seems that keep_prob must be in (0,1], as shown in [here](https://www.tensorflow.org/versions/r1.12/api_docs/python/tf/nn/dropout). But the code run successfully without ValueError.

**Comment by QueeneTam** [comment]
> It seems that keep_prob must be in (0,1], as shown in [here](https://www.tensorflow.org/versions/r1.12/api_docs/python/tf/nn/dropout). But the code run successfully without ValueError.



> It seems that keep_prob must be in (0,1], as shown in [here](https://www.tensorflow.org/versions/r1.12/api_docs/python/tf/nn/dropout). But the code run successfully without ValueError.
Hello, there are some exceptions when I run the code, may you give me some advise about this dropout issue? Mant thanks! Question link:
https://github.com/dstamoulis/single-path-nas/issues/11#issuecomment-549108585

**Comment by qianhuiliu** [comment]
I meet the same question. Have you figured it out?
